### PR TITLE
Move entirely to docker for Classroom

### PR DESCRIPTION
**Pull request by tarebyte**
Since not everyone has a mac, we need to make development as ~~painful~~ painless as possible.

Give it a whirl!

### CODE REVIEW

**Comment by nwoodthorpe** [comment]
My `bin/setup` crashed because I had docker already

```
==> Downloading https://download.docker.com/mac/stable/17661/Docker.dmg
==> Verifying checksum for Cask docker
==> Installing Cask docker
Error: It seems there is already an App at '/Applications/Docker.app'.
Error: nothing to install
Installing docker has failed!
Using terminal-notifier
Using nginx
Homebrew Bundle failed! 1 Brewfile dependencies failed to install.
```

File: script [hunk 1]
```diff
@@ -0,0 +1 @@
+bin
```
Comment (nwoodthorpe) [inline script]: 😆

**Comment by nwoodthorpe** [comment]
Also Docker stores a diff of the container after each command in the dockerfile, it'd be good to try to collapse as many of these commands down. 

E.g. This dockerfile has 1 version with apt-get package lists downloaded, another with apt-get fully updated and mypackages installed., and another with apt-get package lists cleaned out
```
RUN apt-get update
RUN apt-get install mypackages
RUN apt-get clean
```

Whereas this dockerfile has 1 version with all packages installed and no package lists downloaded
```
RUN apt-get update && apt-get install mypackages && apt-get clean
```

**Comment by nwoodthorpe** [comment]
https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/

**Comment by nwoodthorpe** [comment]
Removed docker from brewfile but then it crashed here:

```
Pulling elasticsearch (elasticsearch:1.7.5)...
1.7.5: Pulling from library/elasticsearch
386a066cd84a: Already exists
75ea84187083: Already exists
3e2e387eb26a: Pull complete
eef540699244: Pull complete
1624a2f8d114: Pull complete
7018f4ec6e0a: Pull complete
6ca3bc2ad3b3: Pull complete
424638b495a6: Pull complete
2ff72d0b7bea: Pull complete
8691455bbbd4: Pull complete
7c7a28da409b: Pull complete
6d459d15ba71: Pull complete
c89b17124fad: Pull complete
82d6a34b0f30: Pull complete
Digest: sha256:8dd2aae523b85faa8e5b218a8f6b97c0f848a04c1ad8d6f9fc5e03f69124c38b
Status: Downloaded newer image for elasticsearch:1.7.5
Pulling memcached (memcached:1.4.26-alpine)...
1.4.26-alpine: Pulling from library/memcached
6c123565ed5e: Pull complete
9b764fc1a979: Pull complete
b5b022d021fb: Pull complete
d5f999b5b3c8: Pull complete
c928c95673cc: Pull complete
Digest: sha256:4764f9a4f503546fa11f0faffec6485439a1a9083acfbcf24cc54ea1d56c3eef
Status: Downloaded newer image for memcached:1.4.26-alpine
Starting classroom_elasticsearch_1

ERROR: for elasticsearch  Container sharing network namespace with another container or host cannot be connected to any other network
ERROR: Encountered errors while bringing up the project.

== Command ["docker-compose run web bin/rails db:setup"] failed ==
```

**Comment by tarebyte** [comment]
@nwoodthorpe @wilfriedE the other option I thought of was running our dependencies (pg, elasticsearch, redis etc) as their own containers. Then connecting our locally running ruby server to the containers so that we don't have to give up the really nice workflow we have in development.

Thoughts?

**Comment by nwoodthorpe** [comment]
@tarebyte That sounds good to me, the hassles are usually in getting the dependencies working, not in getting rails working. :+1:

**Pull request closed without merging**